ArcGISRuntimeSDKQt_CppSamples/Scenes/Add3DTilesLayer/Add3DTilesLayer.cpp

```diff
@@ -0,0 +1,129 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "Add3DTilesLayer.h"
+
+#include "AnalysisListModel.h"
+#include "AnalysisOverlay.h"
+#include "AnalysisOverlayListModel.h"
+#include "ArcGISTiledElevationSource.h"
+#include "Basemap.h"
+#include "Camera.h"
+#include "ElevationSourceListModel.h"
+#include "Layer.h"
+#include "LayerListModel.h"
+#include "LocationLineOfSight.h"
+#include "MapTypes.h"
+#include "Ogc3dTilesLayer.h"
+#include "Point.h"
+#include <PortalItem.h>
+#include "Scene.h"
+#include "SceneQuickView.h"
+#include "SpatialReference.h"
+#include "Surface.h"
+#include "TransformationCatalog.h"
+
+#include <MapView.h>
+#include <QDebug>
+#include <QFuture>
+#include <QStandardPaths>
+#include <QString>
+#include <QUrl>
+#include <QVariantMap>
+#include <QMap>
+
+using namespace Esri::ArcGISRuntime;
+
+const QString DATAPATH = QStandardPaths::writableLocation(QStandardPaths::HomeLocation)+ "/ArcGIS/Runtime/Data/";
+
+Add3DTilesLayer::Add3DTilesLayer(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_scene(new Scene(BasemapStyle::ArcGISDarkGray, this))
+{
+  // create a new elevation source from Terrain3D REST service
+  ArcGISTiledElevationSource* elevationSource = new ArcGISTiledElevationSource(
+        QUrl("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"), this);
+  // add projection
+  TransformationCatalog::setProjectionEngineDirectory(DATAPATH + "/pedata");
+  if (TransformationCatalog::projectionEngineDirectory().isEmpty())
+    qWarning() << ("Projection Engine Directory not found at: " + DATAPATH + "/pedata");
+
+  // add the elevation source to the scene to display elevation
+  m_scene->baseSurface()->elevationSources()->append(elevationSource);
+
+  add3DTilesLayer();
+}
+
+Add3DTilesLayer::~Add3DTilesLayer() = default;
+
+void Add3DTilesLayer::init()
+{
+  // Register classes for QML
+  qmlRegisterType<SceneQuickView>("Esri.Samples", 1, 0, "SceneView");
+  qmlRegisterType<Add3DTilesLayer>("Esri.Samples", 1, 0, "Add3DTilesLayerSample");
+}
+
+SceneQuickView* Add3DTilesLayer::sceneView() const
+{
+  return m_sceneView;
+}
+
+// Set the view (created in QML)
+void Add3DTilesLayer::setSceneView(SceneQuickView* sceneView)
+{
+  if (!sceneView || sceneView == m_sceneView)
+    return;
+
+  m_sceneView = sceneView;
+  m_sceneView->setArcGISScene(m_scene);
+
+  emit sceneViewChanged();
+
+  initialize();
+}
+
+void Add3DTilesLayer::initialize()
+{
+  // add an Analysis Overlay
+  m_analysisOverlay = new AnalysisOverlay(this);
+  m_sceneView->analysisOverlays()->append(m_analysisOverlay);
+  // set initial viewpoint
+  setInitialViewpoint();
+}
```

Comment:
We no longer do any analysis so we can remove this
```suggestion
  setInitialViewpoint();
}
```